compiler/rustc_mir_build/src/build/matches/mod.rs

```diff
@@ -2102,13 +2104,18 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
             let (post_guard_block, otherwise_post_guard_block) =
                 self.in_if_then_scope(match_scope, guard_span, |this| {
                     guard_span = this.thir[guard].span;
-                    this.then_else_break(
+                    // Directly invoke the "inner" impl, so that we don't need a
+                    // wrapper function for just this one guard-specific call site.
+                    this.then_else_break_inner(
                         block,
                         guard,
-                        None,
-                        match_scope,
-                        this.source_info(arm.span),
-                        false,
+                        ThenElseArgs {
+                            temp_scope_override: None, // Use `self.local_scope()`
+                            break_scope: match_scope,
+                            variable_source_info: this.source_info(arm.span),
+                            // For guards, `let` bindings are declared separately.
+                            declare_let_bindings: false,
+                        },
```

Comment:
I would prefer if this called `then_else_break` too. Code would be easier to follow if the `inner` function is only called from within a `then_else_break` call imo. Your extra coments are very helpful though, I'd keep those

Reply:
In that case, I’ll undo my small changes to the signature of `then_else_break`, and treat this change as a purely internal refactoring.